app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4234,4 +4242,86 @@ public PlayQueueAdapter getPlayQueueAdapter() {
+     *
+     * How?: In case we are no longer able to write to the surface eg. through rotation/putting in
+     * background we set set a DummySurface. Although it it works on API >= 23 only.
+     * Result: we get a little video interruption (audio is still fine) but we won't get the
+     * 'Unrecoverable player error occurred' error message.
+     *
+     * This implementation is based on:
+     *  'ExoPlayer stuck in buffering after re-adding the surface view a few time #2703'
+     *
+     *  -> exoplayer fix suggestion link
+     *   https://github.com/google/ExoPlayer/issues/2703#issuecomment-300599981
+     */
+    private final class SurfaceHolderCallback implements SurfaceHolder.Callback {
+
+        private final SimpleExoPlayer player;
+        private DummySurface dummySurface;
+
+        SurfaceHolderCallback(final SimpleExoPlayer player) {
+            this.player = player;
+        }
+
+        @Override
+        public void surfaceCreated(final SurfaceHolder holder) {
+            player.setVideoSurface(holder.getSurface());
+        }
+
+        @Override
+        public void surfaceChanged(final SurfaceHolder holder,
+                                   final int format,
+                                   final int width,
+                                   final int height) {
+        }
+
+        @Override
+        public void surfaceDestroyed(final SurfaceHolder holder) {
+            if (dummySurface == null) {
+                dummySurface = DummySurface.newInstanceV17(context, false);
+            }
+            player.setVideoSurface(dummySurface);
+        }
+
+        public void release() {
+            if (dummySurface != null) {
+                dummySurface.release();
+                dummySurface = null;
+            }
+        }
+    }
+
+    private void setupVideoSurface() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) { // >=API23
+            surfaceHolderCallback = new SurfaceHolderCallback(simpleExoPlayer);
+            binding.surfaceView.getHolder().addCallback(surfaceHolderCallback);
+            final Surface surface = binding.surfaceView.getHolder().getSurface();
+            // initially set the surface manually otherwise
+            // onRenderedFirstFrame() will not be called
+            simpleExoPlayer.setVideoSurface(surface);
+        } else {
+            simpleExoPlayer.setVideoSurfaceView(binding.surfaceView);
+        }
+    }
+
+    private void cleanupVideoSurface() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) { // >=API23
+            if (surfaceHolderCallback != null) {
+                if (binding != null) {
+                    binding.surfaceView.getHolder().removeCallback(surfaceHolderCallback);
+                }
+                surfaceHolderCallback.release();
+                surfaceHolderCallback = null;
+            }
+        }
+    }
+    //endregion SurfaceHandlerCallback
```

Comment:
This whole region of code should be moved to another class, `Player` is too big.

Reply:
done